public/components/Flyout/flyouts/createDetector.js

```diff
@@ -0,0 +1,814 @@
+    ];
+    return (
+      <EuiFlexGroup direction="column" gutterSize="s" style={{ paddingTop: '20px' }}>
+        <EuiFlexItem>
+          <EuiTitle size="s">
+            <h3>Features</h3>
+          </EuiTitle>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiPanel grow={false}>
+            <EuiBasicTable
+              items={items}
+              columns={columns}
+              cellProps={() => {
+                return {
+                  textOnly: true,
+                };
+              }}
+            />
+          </EuiPanel>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+}
+
+export class FilterDisplay extends Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    let filter = this.props;
+    if (filter === 'all fields are included') {
+      return (
+        <EuiText>
+          <p className="enabled">-</p>
+        </EuiText>
+      );
+    } else {
+      return <EuiText>{filter}</EuiText>;
+    }
+  }
+}
+const FixedWidthRow = (props) => <EuiFormRow {...props} style={{ width: '150px' }} />;
+
+async function createAndStartDetector(context) {
+  const configs = context.adConfigs;
+  const httpClient = context.httpClient;
+  try {
+    const response = await httpClient.post('../api/alerting/detectors', {
+      configs,
+    });
+    const {
+      data: {
+        ok,
+        response: { _id },
+      },
+    } = response;
+    const detectorId = _id;
+    if (ok) {
+      try {
+        const response = await httpClient.post(`../api/alerting/detectors/${detectorId}/_start`);
+        const {
+          data: {
+            ok,
+            response: { _id },
+          },
+        } = response;
+        if (ok) {
+          context.setFlyout(null);
+          context.renderStartedDetectorFlyout(configs, _id, context.queriesForOverview);
+        }
+      } catch (err) {
+        if (typeof err === 'string') throw err;
+        throw 'There was a problem starting detector';
+      }
+    }
+  } catch (err) {
+    if (typeof err === 'string') throw err;
+    throw 'There was a problem creating detector';
+  }
+}
+
+function warningCallOut(message, color, iconType) {
+  return (
+    <EuiCallOut color={color}>
+      <EuiFlexGroup>
+        <EuiFlexItem grow={false}>
+          <EuiIcon type={iconType} />
+        </EuiFlexItem>
+        <EuiFlexItem>
+          {' '}
+          <EuiTextColor>{message}</EuiTextColor>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiCallOut>
+  );
+}
+
+function triggerCorrectCallOut(context) {
+  if (context.startedDetector) {
+    return detectorCreatedCallOut(context);
+  }
+  let calloutType = 'notValid';
+  if (
+    Object.keys(context.failures).length == 0 &&
+    Object.keys(context.suggestedChanges).length == 0
+  ) {
+    calloutType = 'valid';
+  }
+  for (let [key, value] of Object.entries(context.suggestedChanges)) {
+    if (key === 'detection_interval') {
+      let intervalMinutes = Math.ceil(value[0] / 60000) + 1;
```

Comment:
Why add 1 more minutes as already ceil the value?

Reply:
This was done to account for the time that user might take in changing configurations, and confirming everything so I felt as it was safe to add 1 more minute, I could change this however or be be more clear in my reasoning in adding this through the code

Reply:
How to calculate the detection interval suggested change ? Detection interval should be some stable value from historical data analysis. Here the suggested value is the minimal interval which contains at least one raw data point ? If yes, we should not add 1 more minute. For example, the index has data for every minute, then the minimal suggested interval should be 1 minute, but with this implementation, we suggest user to use 2 minutes.
Should be reasonable to add 1 more minute for window delay.